leftwm/src/config/mod.rs

```diff
@@ -82,19 +96,27 @@ impl WindowHook {
     /// Multiple [`WindowHook`]s might match a `WM_CLASS` but we want the most
     /// specific one to apply: matches by title are scored greater than by `WM_CLASS`.
     fn score_window(&self, window: &Window) -> u8 {
-        let class_score = {
-            let score = self.window_class.is_some()
-                & (self.window_class == window.res_name || self.window_class == window.res_class);
-            u8::from(score)
+        let class_score = match (&self.window_class, &window.res_class) {
+            (Some(wc_re), Some(res_class)) => {
+                if wc_re.is_match(res_class) {
+                    return 1;
+                }
+                return 0;
```

Comment:
There is a `clippy-pedantic` about this:
```rust
warning: boolean to int conversion using if
   --> leftwm/src/config/mod.rs:101:17
    |
101 | /                 if wc_re.is_match(res_class) {
102 | |                     1
103 | |                 } else {
104 | |                     0
105 | |                 }
    | |_________________^ help: replace with from: `u8::from(wc_re.is_match(res_class))`
    |
    = note: `wc_re.is_match(res_class) as u8` or `wc_re.is_match(res_class).into()` can also be valid options
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_to_int_with_if
    = note: `#[warn(clippy::bool_to_int_with_if)]` on by default
```
but maybe if we add a function `raise_score` so it gets even shorter and readable.

Reply:
This might work:
```rust
    fn score_window(&self, window: &Window) -> u8 {
        let class_score = match (&self.window_class, &window.res_class) {
            (Some(wc_re), Some(res_class)) if wc_re.is_match(res_class) => 1,
            _ => 0,
        };

        let title_score = match (&self.window_title, &window.legacy_name) {
            (Some(wt_re), Some(legacy_name)) if wt_re.is_match(legacy_name) => 1,
            _ => 0,
        };

        class_score + 2 * title_score
    }
```

Reply:
> Note that above suggestion should also work and is simpler.

I already suggested the (from my perspective) simplest solution.
```rust
fn score_window(&self, window: &Window) -> u8 {
        let class_score = match (&self.window_class, &window.res_class) {
            (Some(wc_re), Some(res_class)) => u8::from(wc_re.is_match(res_class)),
            _ => 0,
        };

        let title_score = match (&self.window_title, &window.legacy_name) {
            (Some(wt_re), Some(legacy_name)) => u8::from(wt_re.is_match(legacy_name)),
            _ => 0,
        };

        class_score + 2 * title_score
    }
```

Reply:
:man_facepalming: my brain skipped your code block completely, sorry.

Reply:
About the regex only needing to match part of the string, I would use 
```rust
u8::from(regex.replace(input, "") == "")
```

Reply:
@Eskaan Thank you for giving the review to this PR!
I made changes following your suggestions. 

It seems like after these changes, the only thing that is left is figuring out what to do with lookahead/lookbehind expressions. The `regex` crate corresponds to re2 syntax, which doesn't support such kind of expression - [re2 syntax](https://github.com/google/re2/wiki/Syntax). 

We can try to get use of `pcre` bindings, but i guess this variant must  be tested against different platforms, due to compiling of `libpcre2` on crate build.
If re2 syntax seems to be sufficient, mention of the syntax used for this feature must definitely be in docs.

Reply:
Do we have realistic use cases for the lookahead/lookbehind syntax?
I feel like the mention (wiki and comment in the code) of compliance with re2 syntax only should be ok, until someone comes up with an issue that can't be fixed without supporting pcre.